### PR TITLE
Update "friendly string" logic

### PR DESCRIPTION
**Pull request by bummytime**
Lot's of updates to "friendly string" logic in this PR. A picture is worth a thousand words:

3a

## Discussion

In general, I wanted to improve upon the existing numerical recipe found the friendly string logic we extended `Double` with. As it stood, the existing logic could not be localized easily and was problematic to use with currency values.

Some of the specific updates here include:
* Renamed the`Double+Woo.swift` extension's `friendlyString()` method to `humanReadableString()`
* Refactored the existing logic in`Double.humanReadableString()` to make use of NumberFormatter for both small and large numbers so we have better i18n support
* Added `humanReadableString()` to the `NSDecimalNumber+Helpers.swift` extension
* Added a new func to `CurrencyFormatter` → `formatHumanReadableAmount()` which will yield a human-readable string with the ability to both round and not round small numbers to the nearest integer
* In Top Earners, Period Stats, and the Order List screens, I updated all of the appropriate "friendly string" calls to use the new `formatHumanReadableAmount()` func in `CurrencyFormatter`
* Added a lot more unit tests

**Note:** even with all these updates, we are still using the currency settings as defined on the store website.

Fixes: #573 

## Testing

* Make sure all the logic here makes sense
* Run the unit tests — verify they are ✅ 
* Build and run the app — verify everything appears as you would expect (focus on the Order List, and Dashboard screens)
* Build and run the app using different languages — verify everything appears as you would expect

----

@mindgraffiti Please give this a good looksie (sorry about the length here, a lot of it is project files, inline doc comments, and unit tests. 😄 )

@astralbodies if you could put some 👀 on this too, that would be grand!

### CODE REVIEW

**Comment by bummytime** [comment]
Heads up, I just realized the pictures in the original PR description ☝️ are not fully localized. When you run the code here, it will look more like this:

screen shot 2019-01-31 at 3 54 18 pm

😃

File: WooCommerce/Classes/Extensions/NSDecimalNumber+Helpers.swift [hunk 1]
```diff
@@ -1,13 +1,76 @@
...
+    ///  - 1000 becomes "1.0k"
+    ///  - 999999 becomes "1.0m"
+    ///  - 1000000 becomes "1.0m"
+    ///  - 1000000000 becomes "1.0b"
+    ///  - 1000000000000 becomes "1.0t"
+    ///  - 5800199.56 becomes "5.8m"
+    ///
+    /// Note: This helper function does work with negative values as well.
+    ///
+    func humanReadableString(roundSmallNumbers: Bool = true) -> String {
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/NSDecimalNumber+Helpers.swift]: I really like the roundSmallNumbers option!

**Comment by bummytime** [comment]
@mindgraffiti thanks for the review! One comment on this:

> The double extension should be removed. With the extension and convenience methods here, it looks like we are encouraging the use of doubles for currency.

We actually need this to remain still. We have other uses for the `Double` extension besides currency — primarily non-currency fields on stats (for example site visitors and total order count.)

If you want, I can add a bolded note to the double version of `humanReadableString()` reminding us to not use this for currency.

**Comment by mindgraffiti** [comment]
Okay, yeah. Those are perfect use cases for keeping it. A bold note to not abuse Double's `humanReadableString()` would be appreciated.

**Pull request merged**